apps/consulting/posts/scaling-python-banking-edition.md

```diff
@@ -0,0 +1,677 @@
+---
+title: 'Scaling Python: Banking Edition'
+published: March 28, 2023
+author: andrew-fulton
+description: 'In this post, we walk you through how Quansight helped a banking client through the process of scaling a Python data computation in a real life scenario.'
+category: [Scalable Computing, Optimization, Data Engineering]
+featuredImage:
+  src: /posts/scaling-python-banking-edition/header.png
+  alt: 'summing dataframes visualization'
+hero:
+  imageSrc: /posts/hero-paris.webp
+  imageAlt: 'Data visualization of Paris city'
+---
+
+<base target="_blank" />
+
+This post is based off of a talk given at PyData NYC 2022. You can
+watch the talk here: [Scaling Python: Bank
+Edition](https://www.youtube.com/watch?v=tbqgsM6iGng).
+
+Often scaling a distributed computation is easy until it isn't. There
+are currently several tools that will get you off the ground quickly,
+but when the size of your data increases and you are required to scale
+further, the complexity needed for a solution often increases
+considerably as well. At this point, algorithm- and
+infrastructure-dependent requirements will likely preclude
+experimentation and problem solving. While new tools are continuously
+emerging that promise straightforward hyper-scaling, in most
+situations there are at least a few confounding details that have to
+be ironed out manually.
+
+In this post, we walk you through how Quansight helped a banking
+client through this process of scaling Python DataFrame calculations
+in a real-life scenario.
+
+## Problem Overview
+
+For this project, we were tasked with deploying a large scale data processing
+pipeline for running valuation adjustment models using open source tools from
+the PyData ecosystem. We were given 140,000 simulation files, each around 140MB,
+totaling about 20 TB. The data was stored in flat parquet files with columns
+`path`, `Date`, and `value`. Each file contained 50,000 simulations, where each
```

Comment:
@andrewfulton9 -- here is the rewrite of the 'dates into the future' section. Does this accurately capture the shape of the data?

Reply:
Yeah that works for me.